# DSCI 100 Project
## Topic： To what extent does economic vulnerability influence cryptocurrency adoption across countries?

Introduction： As traditional financial and employment systems face increasing strain in many parts of the world, cryptocurrency has emerged as an alternative financial tool. Rather than focusing on a single factor, this project examines whether broader forms of economic vulnerability—such as financial exclusion, labor market conditions, and income levels—are associated with higher levels of crypto adoption.

The goal is to understand whether crypto serves as a substitute for weak traditional systems, particularly in environments where access to banking or stable income is limited.

The data below is based on the World Bank, Chainalysis public information as references and will mainly scrutinize between 2022 to 2025 fluctuations of the data.

### Focus 1:  Do countries with higher inflation rates have higher crypto adoption?

Inflation reflects the degree of monetary instability within an economy, as rising prices reduce the purchasing power of a country’s currency. In such environments, individuals may be more likely to seek alternative stores of value or financial systems, such as cryptocurrency. To examine this relationship, we analyze cross-country data on inflation and crypto adoption.


In [34]:
import pandas as pd
inflation= pd.read_csv("data/Inflation.csv",skiprows=3)
inflation["avg_inflation"] = inflation[["2022", "2023", "2024"]].mean(axis=1)
inflation = inflation[["Country Name", "avg_inflation"]].copy()
inflation["Country Name"] = inflation["Country Name"].str.strip()
inflation

,Country Name,avg_inflation
0,Aruba,NaN
1,Africa Eastern and Southern,7.590817
2,Afghanistan,0.822069
3,Africa Western and Central,5.592821
4,Angola,21.079962
...,...,...
261,Kosovo,6.048095
262,"Yemen, Rep.",NaN
263,South Africa,5.825423
264,Zambia,12.287787


In [29]:
crypto= pd.read_csv("data/crypto_adoption_2024.csv")
crypto

,country,region,overall_rank_2024,sub1_centralized_value,sub2_retail_centralized,sub3_defi_value,sub4_retail_defi
0,India,Central & Southern Asia and Oceania,1,1,1,3,2
1,Nigeria,Sub-Saharan Africa,2,5,2,2,3
2,Indonesia,Central & Southern Asia and Oceania,3,6,6,1,1
3,United States,North America,4,2,12,4,4
4,Vietnam,Central & Southern Asia and Oceania,5,3,3,6,5
...,...,...,...,...,...,...,...
146,Mauritius,Sub-Saharan Africa,147,126,130,150,150
147,Belize,Latin America,148,152,158,123,149
148,Rep. of Congo,Sub-Saharan Africa,149,147,142,151,144
149,Mali,Sub-Saharan Africa,150,88,122,156,152


In [55]:
crypto = crypto.rename(columns={"country": "Country Name"})

In [56]:
inflation_crypto = inflation.merge(crypto, on="Country Name", how="inner")
inflation_crypto

,Country Name,avg_inflation,region,overall_rank_2024,sub1_centralized_value,sub2_retail_centralized,sub3_defi_value,sub4_retail_defi
0,Aruba,NaN,Latin America,146,154,157,138,138
1,Afghanistan,0.822069,Central & Southern Asia and Oceania,31,40,56,8,46
2,Angola,21.079962,Sub-Saharan Africa,139,114,110,149,145
3,Albania,4.566474,Central & Western Europe,109,110,117,111,100
4,Andorra,NaN,Central & Western Europe,111,125,133,103,79
...,...,...,...,...,...,...,...,...
125,United States,5.022888,North America,4,2,12,4,4
126,Uzbekistan,10.343830,Central & Southern Asia and Oceania,33,46,40,30,26
127,South Africa,5.825423,Sub-Saharan Africa,30,19,28,40,35
128,Zambia,12.287787,Sub-Saharan Africa,58,35,54,80,78


In [57]:
inflation_crypto.sort_values(by="overall_rank_2024")

,Country Name,avg_inflation,region,overall_rank_2024,sub1_centralized_value,sub2_retail_centralized,sub3_defi_value,sub4_retail_defi
54,India,5.767071,Central & Southern Asia and Oceania,1,1,1,3,2
89,Nigeria,25.582945,Sub-Saharan Africa,2,5,2,2,3
53,Indonesia,3.353455,Central & Southern Asia and Oceania,3,6,6,1,1
125,United States,5.022888,North America,4,2,12,4,4
123,Ukraine,13.178215,Eastern Europe,6,7,5,5,6
...,...,...,...,...,...,...,...,...
0,Aruba,NaN,Latin America,146,154,157,138,138
85,Mauritius,7.143752,Sub-Saharan Africa,147,126,130,150,150
18,Belize,4.652098,Latin America,148,152,158,123,149
78,Mali,4.962226,Sub-Saharan Africa,150,88,122,156,152


In [66]:
import altair as alt
import pandas as pd

plot_data_filtered = inflation_crypto[[
    "Country Name",
    "avg_inflation",
    "overall_rank_2024",
    "region"
]].dropna()

plot_data_filtered = plot_data_filtered[plot_data_filtered["avg_inflation"] <= 30]

alt.Chart(plot_data_filtered).mark_circle(size=70).encode(
    x="avg_inflation:Q",
    y="overall_rank_2024:Q",
    color="region:N",
    tooltip=["Country Name", "avg_inflation", "overall_rank_2024"]
)

alt.Chart(...)

#### Conclusion of the above findings:
The relationship between inflation and cryptocurrency adoption does not appear strongly linear across countries. While some high-inflation countries show relatively higher adoption, the overall pattern remains dispersed, suggesting that inflation alone is not sufficient to explain differences in crypto adoption.


### Focus 3: Do combined economic vulnerability indicators better explain cryptocurrency adoption?

Since individual variables show limited explanatory power, we now consider a combined model of economic vulnerability. The following analysis examines whether multiple factors together better explain crypto adoption across countries.



In [67]:
GDP= pd.read_csv("data/GDP.csv", skiprows=4)
GDP["avg_gdp"] = GDP[["2022", "2023", "2024"]].mean(axis=1)
GDP=GDP[["Country Name","2022", "2023", "2024","avg_gdp"]]
GDP

,Country Name,2022,2023,2024,avg_gdp
0,Aruba,30975.998912,35718.753119,39498.594129,35397.782054
1,Africa Eastern and Southern,1679.327622,1571.449189,1615.396356,1622.057722
2,Afghanistan,357.261153,413.757895,NaN,385.509524
3,Africa Western and Central,2138.473153,1841.855064,1411.337029,1797.221749
4,Angola,3682.113151,2916.136633,2665.874448,3088.041411
...,...,...,...,...,...
261,Kosovo,5290.950065,6221.206188,7023.065985,6178.407413
262,"Yemen, Rep.",NaN,NaN,NaN,NaN
263,South Africa,6534.248678,6034.272090,6267.186814,6278.569194
264,Zambia,1447.123101,1330.727806,1187.109434,1321.653447


In [68]:
bank=pd.read_csv("data/Bank access.csv",skiprows=3)
bank["bank_access"] = bank["2024"]
bank_clean = bank.dropna(subset=["bank_access"])
bank_clean = bank_clean[["Country Name", "bank_access"]]

In [69]:
inflation_final = inflation[["Country Name", "avg_inflation"]].copy()
inflation_final["Country Name"] = inflation_final["Country Name"].str.strip()

crypto["Country Name"] = crypto["Country Name"].str.strip()

bank = bank[["Country Name", "bank_access"]].copy()
bank["Country Name"] = bank["Country Name"].str.strip()


GDP = GDP[["Country Name", "avg_gdp"]].copy()
GDP["Country Name"] = GDP["Country Name"].str.strip()


merged_all = inflation_final.merge(GDP, on="Country Name", how="inner") \
                            .merge(crypto, on="Country Name", how="inner") \
                            .merge(bank, on="Country Name", how="inner")

merged_all.head()

,Country Name,avg_inflation,avg_gdp,region,overall_rank_2024,sub1_centralized_value,sub2_retail_centralized,sub3_defi_value,sub4_retail_defi,bank_access
0,Aruba,NaN,35397.782054,Latin America,146,154,157,138,138,NaN
1,Afghanistan,0.822069,385.509524,Central & Southern Asia and Oceania,31,40,56,8,46,NaN
2,Angola,21.079962,3088.041411,Sub-Saharan Africa,139,114,110,149,145,NaN
3,Albania,4.566474,9621.868950,Central & Western Europe,109,110,117,111,100,46.069251
4,Andorra,NaN,46176.707752,Central & Western Europe,111,125,133,103,79,NaN


In [70]:
model_data = merged_all[[
    "avg_inflation", 
    "avg_gdp", 
    "bank_access", 
    "overall_rank_2024"
]].dropna()

X = model_data[["avg_inflation", "avg_gdp", "bank_access"]]
y = model_data["overall_rank_2024"]

In [71]:
import sys
!{sys.executable} -m pip install scikit-learn

In [72]:
from sklearn.linear_model import LinearRegression
model = LinearRegression()
model.fit(X, y)
model.coef_, model.intercept_

(array([-2.62434729e-01,  2.49187450e-04, -5.70233451e-01]),
 111.29919996831546)

### Conclusion of the above findings:

The regression results indicate that inflation and financial inclusion play more significant roles in explaining cryptocurrency adoption than GDP alone. In particular, inflation shows a negative relationship with adoption rank (coefficient ≈ −0.26), suggesting that higher inflation is associated with increased crypto usage. Financial inclusion also exhibits a relatively strong effect (coefficient ≈ −0.57), while GDP contributes minimally (coefficient ≈ 0.00025). Overall, the combined model provides a more comprehensive explanation than any individual variable.

A point that is worth mentioning is that while inflation alone does not exhibit a strong linear relationship with crypto adoption, its effect becomes more apparent when controlling for other factors such as GDP and financial inclusion. This suggests that the relationship between inflation and crypto adoption is influenced by multiple interacting variables.


- #### Inflation:
Before combining inflation with other variables, we first examine its individual relationship with cryptocurrency adoption. Based on our initial hypothesis in Focus 1, we test whether inflation alone has a strong explanatory power. The results show that the relationship between inflation and crypto adoption is very weak (R² ≈ 0.004), indicating that inflation alone explains almost none of the variation.

To further evaluate the strength of the relationship, we examine the R² value of the model. 

In [81]:
X1 = merged_all[["avg_inflation"]].dropna()
y1 = merged_all.loc[X1.index, "overall_rank_2024"]

model1 = LinearRegression().fit(X1, y1)
model1.score(X1, y1)

0.003749243332615171

In [82]:
model.score(X, y)

0.06340298076872386

The result for inflation alone is extremely low (R² ≈ 0.004), indicating a very weak relationship. Even in the combined model, the R² remains relatively low (R² ≈ 0.063), suggesting that although the model improves, its overall explanatory power remains limited.
